runtime/compilers/rillv1/parse_model.go

```diff
@@ -248,3 +292,20 @@ func findLineNumber(text string, pos int) int {
 
 	return lineNumber
 }
+
+func fileHash(paths []string) (string, error) {
+	hasher := md5.New()
+	for _, path := range paths {
+		file, err := os.Open(path)
+		if err != nil {
+			return "", err
+		}
```

Comment:
This again becomes the only place with a hard dependency on a real local file system in the `rillv1` package – any chance we could put this under an interface? I realize using `RepoStore.Get` won't do, but perhaps this could be solved by adding a buffered `GetReader(ctx, path) (io.Reader, error)` function to `RepoStore`?

Reply:
Yeah I think it will be better to avoid dependency on local file system, I am assuming it is for near term use cases when `repo` may be stored in some database.
Looking from an implementation POV, in the admin driver I see that all operations happen within a lock. Keeping the lock open till the client closes the reader can be risky so I think we can expose a `FileHash(paths []string)` on the interface ?

Reply:
Yeah the lock is needed to avoid reading the repo while it is being cloned/pulled.

Having a `FileHash` also seems good to me :)